### DISCUSSION TITLE
How can I chat with vision model like "Llama 3.2 90B Vision" from openrouter api ?

### DISCUSSION THREAD

**Discussion by SeqCrafter** [open-webui/open-webui#7734] [Q&A]
I found that when I upload an image, it seems that OpenWeb-UI would send the raw image data instead of the image URL to the API backend. The mechanism also caused a ban of my Grok API because of the massive data in the chat API. 

So the question is how should I do to use an image URL other than image data when I use the external API in OpenWeb-UI?

**Comment by smonux** [comment]
I don't think that's avalaible out of the box. Bear in mind that people may be using it in local computers behind nat, so sending the whole image it's the default use case.

I guess it's doable with a Function Filter, though, by replacing the image literals with the url once stored in a internet reachable location (IIRC there is already a place where files are stored by default and avalaible through http).
